frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts

```diff
@@ -17,42 +18,51 @@ import { CHART_STYLE } from "metabase/visualizations/echarts/cartesian/constants
 
 import { buildEChartsScatterSeries } from "../scatter/series";
 import { buildEChartsWaterfallSeries } from "../waterfall/series";
+import type { WaterfallChartModel } from "../waterfall/types";
 import { getSeriesYAxisIndex } from "./utils";
 
-const buildEChartsLabelOptions = (
+export function getDataLabelFormatter(
```

Comment:
I separate this from `buildEChartsLabelOptions` so that we can reuse it for the waterfall chart.